### ISSUE TITLE
Help to download Blessings 祖先保佑 full series

### ISSUE THREAD

**Issue by ds2003** [0x776b7364/toggle.sg-download#54]
Hi

Currently, I'm using download_toggle_video2.py to d/l newer toggle series and the old download_toggle_video.py (in which I have to manually choose the "hlstv_hd/f/a.m3u8" option) for older series.

However, for Blessings series, it seems like there are 3 groups of links:

1) Using download_toggle_video2.py , I'm able to d/l ALL the episodes' english subtitles and the following episodes: ep2, ep3, ep4, ep11, ep12, ep13, ep14, ep16, ep18 & ep20
2) Using download_toggle_video.py , I have to manually choose the a.m3u8 option from the list of options and get the mp4 file in its 6 digits filename (e.g. ep1 was downloaded as 301012.mp4)
Eps able to be downloaded: ep1, ep10, ep15, ep17 & ep19
3) Wasn't able to d/l the following episodes using either of the py files,
ep5, ep6, ep7, ep8, ep9
I was able to play these 5 episodes on toggle.sg

Following are the 20 episodes links from toggle:
https://video.toggle.sg/en/series/blessings/ep1/301012
https://video.toggle.sg/en/series/blessings/ep2/301014
https://video.toggle.sg/en/series/blessings/ep3/301015
https://video.toggle.sg/en/series/blessings/ep4/301016
https://video.toggle.sg/en/series/blessings/ep5/301017
https://video.toggle.sg/en/series/blessings/ep6/301877
https://video.toggle.sg/en/series/blessings/ep7/301937
https://video.toggle.sg/en/series/blessings/ep8/302047
https://video.toggle.sg/en/series/blessings/ep9/302048
https://video.toggle.sg/en/series/blessings/ep10/304207
https://video.toggle.sg/en/series/blessings/ep11/302623
https://video.toggle.sg/en/series/blessings/ep12/302841
https://video.toggle.sg/en/series/blessings/ep13/302842
https://video.toggle.sg/en/series/blessings/ep14/302843
https://video.toggle.sg/en/series/blessings/ep15/303304
https://video.toggle.sg/en/series/blessings/ep16/303305
https://video.toggle.sg/en/series/blessings/ep17/303306
https://video.toggle.sg/en/series/blessings/ep18/303307
https://video.toggle.sg/en/series/blessings/ep19/303805
https://video.toggle.sg/en/series/blessings/ep20/303806

For the manual options, I've tried all of them for ep5 to 9 but only those for a.wvm works. Those pointing to a.m3u8 all error'ed out saying the file did not exist.

Can anyone see if they have a way to download ep5 to 9 as mp4 files (i.e d/l the m3u8 files)?
Thanks.

**Comment by 0x776b7364** [comment]
If the episodes were recently published online, I'd wait a few weeks for them to be available through the script. I cannot say for sure how Toggle works, but local shows are almost always 'available' in the end.

**Comment by ds2003** [comment]
No, they're not new. The series was aired in 2014, though I have to admit they are currently re-airing the series in the off-peak periods cause "season 2" is going to be aired next month Oct 2018.

Not sure if they had "re-published" the toggle episodes, though the series is 4 years old, but the site did state the episodes were only "published" last week.

Okay, thanks, I'll wait for a few weeks and try again. Thanks!!

**Comment by hexapling** [comment]
I had no problems downloading using Youtube-dl or its GUI, Youtube-DLG:

https://mrs0m30n3.github.io/youtube-dl-gui/

**Comment by ds2003** [comment]
Just tried Youtube-dl GUI version as recommended on ep5, got the following error (status of the download was "Warning").

[10/03/18 14:44:02] WARNING: Failed to download DASH_HBBTV MPD manifest: HTTP Error 404: Not Found
[10/03/18 14:44:02] WARNING: hlsnative has detected features it does not support, extraction will be delegated to ffmpeg

@hexapling did you try d/l ep 5 to 9 which were the episodes I was unable to d/l (and still not able to even though the repeat telecast has ended last week).

**Comment by hexapling** [comment]
I'm terribly sorry, @ds2003 I only tested the first couple of episodes. Didn't know 5 to 9 isn't working. Hope you'll find a solution soon!

**Comment by ds2003** [comment]
Very strange.... Have tried all sites that claimed to download toggle.sg videos. They can download other toggle local videos but not Blessings ep 5 to 9.... Just these 5 episodes.
Frustrating also is that the downloadable wvm file only plays up to the first minute, even though the downloaded file is the whole episode, probably due to the DRM protection.

If any techie can help look into this, I would really appreciate this!!

BTW, just discovered that Wonderfox video conversion programs can actually download toggle videos!! But even this program can't download Blessings ep 5 to 9?!?!

**Comment by 0x776b7364** [comment]
So.. @ds2003 have you verified if the script works for your episodes 5-9? As I've indicated in the README, other tools may work better for downloads, so feel free to use them!
